encoding/length.go

```diff
@@ -0,0 +1,94 @@
+package encoding
+
+import (
+	"encoding/binary"
+	"io"
+
+	"github.com/xichen2020/eventdb/generated/proto/encodingpb"
+	"github.com/xichen2020/eventdb/x/unsafe"
+)
+
+type lengthStringIterator struct {
+	reader    io.Reader
+	meta      encodingpb.StringMeta
+	container []byte
+	curr      string
+	err       error
+	closed    bool
+	zero      uint64
+}
+
+func newLengthStringIterator(reader io.Reader, meta encodingpb.StringMeta) ForwardStringIterator {
```

Comment:
I wanted the iterator method signatures and internal data to be consistent so it's less mental overhead. Otherwise, the iterator ctor will all be diff and I'll have to change the ctor signature and internal data every time I add/remove meta info.

Reply:
Discussed offline, decided to go with the approach that passes the minimum amount of information needed.